core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -112,6 +112,21 @@ sealed abstract class Parser[+A] {
   def ? : Parser[Option[A]] =
     Parser.oneOf(Parser.map(this)(Some(_)) :: Parser.Impl.optTail)
 
+  /** If this parser fails to parse its input with an epsilon error,
+    * try the given parser instead.
+    *
+    * If this parser fails with an arresting error, the next parser
+    * won't be tried.
+    *
+    * Backtracking may be used on the left parser to allow the right
+    * one to pick up after any error, resetting any state that was
+    * modified by the left parser.
+    *
+    * This method is similar to Parser#orElse but returns Either.
+    */
+  def or[B](pb: Parser[B]): Parser[Either[A, B]] =
```

Comment:
one challenge is the convention that the Right is "right". So, I might say, returning `Parser[Either[B, A]]` instead.

That said, I know that that is also the case that lexically the left side has type A and the right side has type B, so it seems that Left and Right should match...

I'm somewhat inclined to make a new ADT that has First and Second, which side steps this issue, but then that requires users to use those.

Finally, I'm a bit concerned with the name since it is so close to orElse. I know the method is similar to orElse, so maybe that's not a problem.

what about eitherOr? That is an expression people use in English. It is longer, but it does make it clearer which one is adding Either.

Reply:
Does it have to be an infix method?  `P.either(pa, pb)` feels natural to me.  There'd need to be an `either0` or an `either1`, depending on the fate of #116.

I think we this inlined a few places in http4s already. I'm in favor of the general concept.

Reply:
I actually do like `P.either` actually... but it does not solve the question of Left result being the default, when Right is usually the canonical value.

PS: cats should just add a Result type with Err and Ok and copy rust. This Either business is a mistake when combined with one side usually/often having a bias....

Reply:
(and yes, I remember Xor, and I remember the debates about making Either right biased... I think Either should be right biased, but Result a-la rust is a better design than Xor).

Reply:
actually, isn't this maybe a PR for cats in Alternative of MonoidK + Functor (there isn't a type like that is there?)

Reply:
Hi! sorry for the late reply! A lot of good comments here that left me with no really good answer actually.

Initially, I had `P.either` but then I thought that this would probably be used more as infix, similar to `orElse`, and kept it lexical for Left and Right. @johnynek very good point about the `right` bias and I think it should be `Parser[Either[B, A]]`. I'm lacking an alternative name to `or` that conveys better "error" meaning to the Left. Open for suggestions...

> cats should just add a Result type with Err and Ok and copy rust. This Either business is a mistake when combined with one side usually/often having a bias....

+1

> actually, isn't this maybe a PR for cats in Alternative or MonoidK + Functor (there isn't a type like that is there?)

not sure... maybe? but isn't `orElse` more in line with `Alternative`?

Reply:
would `recover` be completely off?